spec/lib/cleanup/destroy_unused_providers_spec.rb

```diff
@@ -0,0 +1,131 @@
+require 'rails_helper'
+require './lib/cleanup/destroy_unused_providers'
+
+RSpec.describe DestroyUnusedProviders do
+  let(:stdout) { StringIO.new }
+  let(:stdin) { StringIO.new }
+  let(:service_provider) { create(:service_provider) }
+  let(:issuer) { service_provider.issuer }
+
+  subject(:destroy_unused_providers) { described_class.new([issuer], stdout:, stdin:) }
+
+  before do
+    allow(stdin).to receive(:puts)
+  end
+
+  describe '#init' do
+    it 'instantiates the stdin' do
+      expect(subject.stdin).to eq stdin
+    end
+
+    it 'instantiates the stdout' do
+      expect(subject.stdout).to eq stdout
+    end
+
+    it 'creates the destroy_list' do
+      expect(subject.destroy_list.first.class).to eq DestroyableRecords
+    end
+  end
+
+  describe '#run' do
+    describe 'when an issuer does not match an existing service_provider' do
+      let(:issuer) { 'issuer-not-in-db' }
+
+      it 'outputs some helpful information' do
+        expect(stdout).to receive(:puts).with("Issuer #{issuer} is not associated with a service provider.")
```

Comment:
So my personal favorite part of passing `StringIO`s here is that we no longer need to stub, we can just check the result directly

(this will need to be moved *after* the run call but the gist is the same)

```suggestion
        expect(stdout.string).to include("Issuer #{issuer} is not associated with a service provider.")
```

Reply:
ah cool!

Reply:
done! thanks for your help [`00325ac` (#8905)](https://github.com/18F/identity-idp/pull/8905/commits/00325ac141bdc991836fff3a8782157a9b484287)